invenio_rdm_migrator/state.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Migrator is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Base state module."""
+
+from abc import ABC, abstractclassmethod
+from pathlib import Path
+
+from sqlalchemy import create_engine, insert, select, update
+from ..utils import ts
+
+
+class State:
+    """Migration state."""
+
+    def __init__(self, dbpath, meta, validators=None):
```

Comment:
The constructor will receive a list of "models" to track, and those tables are going to be binded to a meta defined in the `State` class. Since we want to avoid going into registration a-la-blueprints/endpoint, the models will be hardcoded on the `Runner` when the state is instantiated. This way all models will always be tracked.

This also helps us avoid problems when loading partial databases and then overwritting on `save`.

```suggestion
    def __init__(self, dbpath, models, validators=None):
```